src/modules/office/OfficeView.js

```diff
@@ -0,0 +1,207 @@
+    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
+    this.state = {
+      dataSource: ds.cloneWithRows(offices)
+    };
+  }
+
+  changePositionPager = (position) => {
+    this.props.officeStateActions.changePosition(position);
+  }
+
+  selectOffice = (office) => {
+    console.log('Selected office: ', office);
+    this.props.officeStateActions.selectOffice(office);
+  }
+
+  renderRow = (rowData, section, index) => {
+    // Show pageIndicator and button for Android on the row because the function
+    // 'onChangeVisibleRows' does not work for Android
+    let androidView = (Platform.OS === 'android')
+      ? (<View style={styles.buttonsContainer}>
+          <PageIndicator pageCount={offices.length}
+            selectedIndex={+index}
+            style={styles.pageIndicator} />
+          <Button
+              text="What's for lunch?"
+              buttonStyle={theme.buttons.primary}
+              textStyle={theme.fonts.primary}
+              action={() => this.selectOffice(offices[index])} />
+        </View>)
+      : (<View/>);
+
+    return (
+      <View style={styles.cityCard}>
+        <TouchableOpacity onPress={() => this.selectOffice(offices[index])}>
+          <Image source={{uri: rowData.picture}} style={styles.image} />
+        </TouchableOpacity>
+        <Text style={[theme.fonts.h1, styles.title]}>
+          {rowData.city.toUpperCase()}
+        </Text>
+        {androidView}
+      </View>
+    );
+  }
+
+ // This method is currently only working on iOS but not on Android
+  onChangeVisibleRows = (visibleRows) => {
+    const visibleRowNumbers = Object.keys(visibleRows.s1).map((row) => parseInt(row));
+    if (visibleRowNumbers.length === 2) {
+      // visible row is visibleRowNumbers[0]
+      // but in the case of the last item it is visibleRowNumbers[1]
+      if (visibleRowNumbers[1] === (offices.length - 1)) {
+        this.changePositionPager(visibleRowNumbers[1]);
+
+      } else {
+        this.changePositionPager(visibleRowNumbers[0]);
+      }
+    }
+    if (visibleRowNumbers.length === 3) {
+      // visible row is visibleRowNumbers[1]
+      this.changePositionPager(visibleRowNumbers[1]);
+    }
+  }
+
+  render() {
+    var spinner = this.props.loading
+      ? (<ActivityIndicator style={styles.spinner} size='large' color='white'/>)
+      : (<View/>);
+
+    // Hide pageIndicator and button for Android because the function
+    // 'onChangeVisibleRows' does not work for Android
+    let iosView = (Platform.OS === 'ios')
+      ? (<View style={styles.buttonsContainer}>
+          <PageIndicator pageCount={offices.length}
+            selectedIndex={this.props.position}
+            style={styles.pageIndicator} />
+          <Button
+              text="What's for lunch?"
+              buttonStyle={theme.buttons.primary}
+              textStyle={theme.fonts.primary}
+              action={() => this.selectOffice(offices[this.props.position])} />
+         </View>)
```

Comment:
This seems very similar to the androidView a bit higher, do you think it's worthwhile to combine them?